polonius-engine/src/output/datafrog_opt.rs

```diff
@@ -418,11 +396,14 @@ pub(super) fn compute<T: FactTypes>(
             //
             // subset_placeholder(Origin1, Origin3, Point) :-
             //     subset_placeholder(Origin1, Origin2, Point),
-            //     subset(Origin2, Origin3, Point).
-            subset_placeholder.from_join(
+            //     subset(Origin2, Origin3, Point),
+            //     Origin1 != Origin3.
+            subset_placeholder.from_join_filtered(
                 &subset_placeholder_o2p,
                 &subset_o1p,
-                |&(_origin2, point), &origin1, &origin3| (origin1, origin3, point),
+                |&(_origin2, point), &origin1, &origin3| {
+                    (origin1 != origin3).then(|| (origin1,  origin3, point))
```

Comment:
```suggestion
                    (origin1 != origin3).then(|| (origin1, origin3, point))
```
startup idea: rustfmt-as-a-service